File: Day03Assignment/src/com/sonata/Employee.java
package com.sonata;

	public class Employee extends MyOwnException implements MyInterface{
		int  empId;
		String empName;
		double empSal;
		
		public Employee(int id, String name, double sal, String a) {
			super(a);
			this.empId = id;
			this.empName = name;
			this.empSal = sal;
			
		}
		
		public void display() {
			System.out.println("Employee id is : "+empId);
			System.out.println("Employee Name : "+empName);
			System.out.println("Employee Salary : "+empSal);
		}


		@Override
		public void saCal() throws MyOwnException 
		{
			
			double yearlySal = empSal * 12;
			
				if(yearlySal < 300000)
				{
					throw new MyOwnException(" lessthan 1 lakh");
				}else 
				{
					System.out.println("greaterrthan 1 lakh");
				}
		
		}
		public static void main(String[] args) throws MyOwnException 
		{
			Employee emp = new Employee(21275, "Karthik", 26584 , "hello");
			emp.display();
			emp.saCal();
			
		}
		
	}

File: TypeScript/Employee.js
"use strict";
exports.__esModule = true;
exports.Employee = void 0;
var Employee = /** @class */ (function () {
    function Employee(i, n, s) {
        this.empID = 123;
        this.empName = "karthik";
        this.empSal = 1275;
        this.empID = i;
        this.empName = n;
        this.empSal = s;
    }
    Employee.prototype.display = function () {
        console.log(this.empID);
        console.log(this.empName);
        console.log(this.empSal);
    };
    return Employee;
}());
exports.Employee = Employee;
var e1 = new Employee(124, "beast", 345);
e1.display();
